### PR TITLE
refactor: remove isCLIAPIVersionOne related codes

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7191

### CODE REVIEW

**Review by innobead — DISMISSED** [review]

**Comment by innobead** [comment]
The PR check failed due to a lint issue. Let's fix it before merging.

**Comment by derekbit** [comment]
> The PR check failed due to a lint issue. Let's fix it before merging.

Fixed. Thank you.

File: controller/engine_controller.go [hunk 1]
```diff
@@ -360,15 +350,6 @@ func (ec *EngineController) syncEngine(key string) (err error) {
 	}
 
 	// For incompatible engine, skip starting engine monitor and clean up fields when the engine is not running
```
Comment (innobead) [inline controller/engine_controller.go:352]: Remove this outdated message.

**Review by innobead — CHANGES_REQUESTED** [review]

File: controller/engine_controller.go (hunk 1 above)
Comment (derekbit) [reply controller/engine_controller.go]: Removed. Thank you.

**Review by shuo-wu — APPROVED** [review]

File: controller/instance_manager_controller.go [hunk 2]
```diff
@@ -940,14 +940,14 @@ func formatInstanceMessage(im *longhorn.InstanceManager) string {
 		}
 	}
 
-	if len(im.Status.Instances) > 0 {
+	if len(im.Status.Instances) > 0 { // nolint: staticcheck
```
Comment (PhanLe1010) [inline controller/instance_manager_controller.go:943]: Any idea why we need to have `nolint` here?

**Review by PhanLe1010 — APPROVED** [review]

File: controller/instance_manager_controller.go (hunk 2 above)
Comment (PhanLe1010) [reply controller/instance_manager_controller.go:943]: Oh. ok they are deprecated fields as @james-munson pointed out here https://github.com/longhorn/longhorn-manager/pull/3069

**Review by innobead — APPROVED** [review]

**Pull request merged**